### PR TITLE
Editorial: OrdinaryGetOwnPropery does not return a completion

### PR DESCRIPTION
**Pull request by annevk**

### CODE REVIEW

**Comment by annevk** [comment]
(This is largely my fault since I added both ! while fixing other things (and neglected the other calls).)

File: spec.html [hunk 1]
```diff
@@ -7598,7 +7598,7 @@
         <emu-alg>
           1. Let _args_ be the arguments object.
           1. If _P_ is `"caller"`, then
-            1. Let _desc_ be ! OrdinaryGetOwnProperty(_args_, _P_).
+            1. Let _desc_ be OrdinaryGetOwnProperty(_args_, _P_).
             1. If IsDataDescriptor(_desc_) is *true*, return *true*.
```
Comment (ljharb) [inline spec.html]: based on this usage, `_desc_` could be an abrupt completion and you're not checking it - I'm confused why `!` isn't appropriate here?

Comment (annevk) [inline spec.html]: I might not understand completion records completely, but OrdinaryGetOwnProperty only ever returns undefined or a Property Descriptor. How can that be abrupt?

Comment (ljharb) [inline spec.html]: ah, you're right - it can't be. Why not leave the `!` then, to make it clear that it can't be abrupt?

Comment (annevk) [inline spec.html]: Because ! unwraps a completion record, but if a completion record is never returned, it would segfault or some such, no?

Comment (annevk) [inline spec.html]: Oh I have that wrong. Only if a completion record is returned it will unwrap, otherwise it'll leave the return value be. Ugh.

Comment (bterlson) [inline spec.html]: Per 5.2 all abstract operations return completion records. Either ! or ? must be used to unwrap a value (although we retain for now the semantics where we can implicitly treat a completion record as its unwrapped value in certain circumstances).

**Pull request closed without merging**